addons/website_customer/controllers/main.py

```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
     @http.route([
         '/customers',
         '/customers/page/<int:page>',
-        '/customers/country/<int:country_id>',
-        '/customers/country/<country_name>-<int:country_id>',
-        '/customers/country/<int:country_id>/page/<int:page>',
-        '/customers/country/<country_name>-<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>',
-        '/customers/tag/<tag_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<int:country_id>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>',
-        '/customers/tag/<tag_id>/country/<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>/page/<int:page>',
+        '/customers/country/<model("res.country"):country>',
+        '/customers/country/<model("res.country"):country>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>',
+        '/customers/industry/<model("res.partner.industry"):industry>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>/page/<int:page>',
     ], type='http', auth="public", website=True)
-    def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
-        Country = request.env['res.country']
+    def customers(self, country=None, industry=None, page=0, **post):
         Tag = request.env['res.partner.tag']
         Partner = request.env['res.partner']
         partner_name = post.get('search', '')
 
         domain = [('website_published', '=', True), ('assigned_partner_id', '!=', False)]
         if partner_name:
             domain += [
-                '|',
+                '|', '|',
                 ('name', 'ilike', post.get("search")),
-                ('website_description', 'ilike', post.get("search"))
+                ('website_description', 'ilike', post.get("search")),
+                ('industry_id.name', 'ilike', post.get("search")),
             ]
 
+        tag_id = post.get('tag_id', False)
         if tag_id:
             tag_id = unslug(tag_id)[1] or 0
             domain += [('website_tag_ids', 'in', tag_id)]
 
+        # group by industry, based on customers found with the search(domain)
+        industries = Partner.sudo().read_group(domain, ["id", "industry_id"], groupby="industry_id", orderby="industry_id")
+        industry_count = Partner.sudo().search_count(domain)
```

Comment:
I know it's the same pattern as below, but `industry_count` is a confusing name for a variable holding a partner count